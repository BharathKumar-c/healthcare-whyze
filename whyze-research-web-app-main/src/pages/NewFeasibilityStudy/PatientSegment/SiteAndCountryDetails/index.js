import React from 'react';
import PropTypes from 'prop-types';
import { Col, Row } from 'antd';
import { Text } from '../../../../components/basic';
import { InfoCircleFilled } from '@ant-design/icons';
import { Popover } from '../../../../components/hoc';

export default function SiteAndCountryDetails({
  icon,
  value,
  title,
  isInfo,
  tab,
}) {
  const content = (
    <div className="content">
      Additional Whyze patients in site catchment areas, but not yet connected
      to these sites.
    </div>
  );
  return (
    <div className="site-container">
      <Row align="middle" wrap style={{ marginBottom: '10px' }}>
        <Col span={12}>
          <div
            className={
              isInfo ? 'site-container-icon-grey-bg' : 'site-container-icon'
            }
          >
            {icon}
          </div>
        </Col>
        <Col span={12}>
          <Text className="no-of-patients" text={value} />
        </Col>
      </Row>
      <Row align="middle">
        <Col span={isInfo ? 22 : 24}>
          <Text
            className={
              tab === 'feasibility' || tab === 'location'
                ? 'patient-segment-title'
                : 'patient-segment-title-small'
            }
            text={title}
          />
        </Col>
        {isInfo && (
          <Col span={2}>
            <Popover
              id="not-connected-patients"
              trigger="hover"
              content={content}
              placement="bottomRight"
            >
              <InfoCircleFilled />
            </Popover>
          </Col>
        )}
      </Row>
    </div>
  );
}

SiteAndCountryDetails.propTypes = {
  value: PropTypes.number,
  title: PropTypes.string,
  icon: PropTypes.node,
  isInfo: PropTypes.bool,
  tab: PropTypes.string,
};

SiteAndCountryDetails.defaultProps = {
  value: '',
  title: '',
  icon: null,
  isInfo: false,
  tab: '',
};
