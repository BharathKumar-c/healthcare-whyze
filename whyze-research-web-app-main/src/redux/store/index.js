import { configureStore } from '@reduxjs/toolkit';
import reducer from '../reducer/combinedReducer';

const store = configureStore({
  reducer: reducer,

  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: false,
    }),
});

export default store;
