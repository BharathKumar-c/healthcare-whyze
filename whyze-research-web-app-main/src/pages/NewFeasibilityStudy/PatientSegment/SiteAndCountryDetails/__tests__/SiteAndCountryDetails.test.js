import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../../../../../redux/store';
import SiteAndCountryDetails from '..';
import React from 'react';

function renderWithContext(element) {
  const { container, getByText } = render(
    <Provider store={store}>{element}</Provider>,
  );
  return { container, store, getByText };
}
describe('create tests for SelectedCountry', () => {
  test('create snapshots', () => {
    const { container } = renderWithContext(<SiteAndCountryDetails />);
    expect(container).toMatchSnapshot();
  });
});
