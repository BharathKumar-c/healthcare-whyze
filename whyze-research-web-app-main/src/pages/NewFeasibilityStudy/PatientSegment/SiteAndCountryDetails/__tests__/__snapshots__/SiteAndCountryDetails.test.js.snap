// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create tests for SelectedCountry create snapshots 1`] = `
<div>
  <div
    class="site-container"
  >
    <div
      class="ant-row ant-row-middle"
      style="margin-bottom: 10px;"
    >
      <div
        class="ant-col ant-col-12"
      >
        <div
          class="site-container-icon"
        />
      </div>
      <div
        class="ant-col ant-col-12"
      >
        <span
          class="ant-typography no-of-patients"
        />
      </div>
    </div>
    <div
      class="ant-row ant-row-middle"
    >
      <div
        class="ant-col ant-col-24"
      >
        <span
          class="ant-typography patient-segment-title-small"
        />
      </div>
    </div>
  </div>
</div>
`;
